TwitterStats/run.ps1

```diff
@@ -0,0 +1,225 @@
+            Invoke-TwitterRestMethod -ResourceURL 'https://api.twitter.com/1.1/statuses/retweets/509457288717819904.json' -RestVerb 'GET' -Parameters @{} -OAuthSettings $OAuth
+            Invoke-TwitterRestMethod -ResourceURL 'https://api.twitter.com/1.1/statuses/show.json' -RestVerb 'GET' -Parameters @{'id'='123'} -OAuthSettings $OAuth 
+            Invoke-TwitterRestMethod -ResourceURL 'https://api.twitter.com/1.1/statuses/destroy/240854986559455234.json' -RestVerb 'GET' -Parameters @{} -OAuthSettings $OAuth 
+            Invoke-TwitterRestMethod -ResourceURL 'https://api.twitter.com/1.1/statuses/update.json' -RestVerb 'POST' -Parameters @{'status'='@FollowBot'} -OAuthSettings $OAuth 
+            Invoke-TwitterRestMethod -ResourceURL 'https://api.twitter.com/1.1/direct_messages.json' -RestVerb 'GET' -Parameters @{} -OAuthSettings $OAuth 
+            Invoke-TwitterRestMethod -ResourceURL 'https://api.twitter.com/1.1/direct_messages/destroy.json' -RestVerb 'POST' -Parameters @{'id' = '559298305029844992'} -OAuthSettings $OAuth 
+            Invoke-TwitterRestMethod -ResourceURL 'https://api.twitter.com/1.1/direct_messages/new.json' -RestVerb 'POST' -Parameters @{'text' = 'hello, there'; 'screen_name' = 'ruminaterumi' } -OAuthSettings $OAuth 
+            $mediaId = Invoke-TwitterMEdiaUpload -MediaFilePath 'C:\Books\pic.png' -ResourceURL 'https://upload.twitter.com/1.1/media/upload.json' -OAuthSettings $OAuth 
+            Invoke-TwitterRestMethod -ResourceURL 'https://api.twitter.com/1.1/statuses/update.json' -RestVerb 'POST' -Parameters @{'status'='FollowBot'; 'media_ids' = $mediaId } -OAuthSettings $OAuth 
+
+     #>
+         [CmdletBinding()]
+	     [OutputType('System.Management.Automation.PSCustomObject')]
+         Param(
+                [Parameter(Mandatory)]
+                [string]$ResourceURL,
+                [Parameter(Mandatory)]
+                [string]$RestVerb,
+                [Parameter(Mandatory)]
+                $Parameters,
+                [Parameter(Mandatory)]
+                $OAuthSettings
+
+                )
+
+          process{
+              try{
+
+                    $AuthParams = @{}
+                    $AuthParams.Add('HttpEndPoint', $ResourceURL)
+                    $AuthParams.Add('RESTVerb', $RestVerb)
+                    $AuthParams.Add('Params', $Parameters)
+                    $AuthParams.Add('OAuthSettings', $OAuthSettings)
+                    $AuthorizationString = Get-OAuth -AuthorizationParams $AuthParams                 
+                    $HTTPEndpoint= $ResourceURL
+                    if($Parameters.Count -gt 0)
+                    {
+                        $HTTPEndpoint = $HTTPEndpoint + '?'
+                        $Parameters.Keys | % { $HTTPEndpoint = $HTTPEndpoint + $_  +'='+ [System.Net.WebUtility]::UrlEncode($Parameters.Item($_)).Replace('+','%20') + '&'}

```

Comment:
@devlead good catch.  I borrowed this function directly.  I have made the suggested change.